ci/scripts/test.sh

```diff
@@ -23,14 +23,22 @@ source_dir=${1}
 
 # Python and PyArrow are required for C Data Interface tests.
 if [ -z "${PYTHON:-}" ]; then
-  if type python3 > /dev/null 2>&1; then
+  if type python3 >/dev/null 2>&1; then
     export PYTHON=python3
   else
     export PYTHON=python
   fi
 fi
-${PYTHON} -m pip install pyarrow find-libpython
-PYTHONNET_PYDLL=$(${PYTHON} -m find_libpython)
+if [ "$(uname)" = "Linux" ]; then
+  ${PYTHON} -m venv create arrow-dotnet-dev
```

Comment:
I want to use venv on all platforms but venv doesn't work on macOS and Windows...
We may revisit this later...